src/freenet/clients/http/QueueToadlet.java

```diff
@@ -1353,7 +1355,17 @@ else if(secondFinalized && !firstFinalized)
 							result = Fields.compare(firstProgress, secondProgress);
 						}
 					} else if (sortBy.equals("lastActivity")) {
-						result = Fields.compare(firstRequest.getLastActivity(), secondRequest.getLastActivity());
+                        Date a = firstRequest.getLastActivity();
+                        Date b = secondRequest.getLastActivity();
+                        if(a == null) a = new Date(0);
```

Comment:
> This code is duplicated. Thoughts on moving it out to a function which allows null?
> EDIT: What about adding Date support to Fields.compare()?

Fixed by 88c73464bd82234b05ca6113481d9ad9131c4ed3

> The coding standards require a space between the if and (. They also do not allow the omission of braces here.

Fixed by 9e0d129a4c2eaafe3872a63e1da0f154667b8574

> Why Date(0) on null?

Added cb4b91dc904fa36e61510676f5aa5fedb3f7bec5 to explain this.
Also notice that the Date objects can be null here because if the user was on a purge-db4o testing release before his serialized data would not contain the Date fields. That part was not added to the comment in the above commit since the Fields class is unrelated to the "Last activity" stuff and thus shouldn't document its quirks.